src/test/java/com/snowflake/kafka/connector/internal/StageFilesProcessorTest.java

```diff
@@ -457,9 +491,28 @@ void fileProcessor_WillCleanHistoryEntries_OlderThanOneHour() {
     // check if these old not matched files have been removed from history and do not result in
     // memory leak
     assertThat(register.currentProcessorContext.ingestHistory).isEmpty();
+    initAssert.run();
+  }
+
+  private Runnable configureInitialState() {
```

Comment:
It's a little bit confusing to me as the test setup and result verification is mixed. Wdyt about splitting it into two separate methods?

Reply:
Then it wouldn't have to return Runnable which suggests running it from a different thread.